extension/js/common/api/key-server/wkd.ts

```diff
@@ -27,59 +29,60 @@ export class Wkd extends Api {
-      //    the user typically gets the key loaded from composing anyway
-      // the proper fix would be to run encodeZBase32 through background scripts
-      return { keys: [], errs: [] };
+      return [];
     }
     const lowerCaseRecipientDomain = recipientDomain.toLowerCase();
     const directDomain = lowerCaseRecipientDomain;
     const timeout = (this.usesKeyManager && lowerCaseRecipientDomain === this.domainName) ? 10 : 4;
     const advancedDomainPrefix = (directDomain === 'localhost') ? '' : 'openpgpkey.';
-    const hu = opgp.util.encodeZBase32(await opgp.crypto.hash.digest(opgp.enums.hash.sha1, Buf.fromUtfStr(user.toLowerCase())));
+    const hashed = await window.crypto.subtle.digest('SHA-1', Buf.fromUtfStr(user.toLowerCase()));
+    const hu = this.encodeZBase32(new Uint8Array(hashed));
     const directHost = (typeof this.port === 'undefined') ? directDomain : `${directDomain}:${this.port}`;
     const advancedHost = `${advancedDomainPrefix}${directHost}`;
     const userPart = `hu/${hu}?l=${encodeURIComponent(user)}`;
     const advancedUrl = `https://${advancedHost}/.well-known/openpgpkey/${directDomain}`;
     const directUrl = `https://${directHost}/.well-known/openpgpkey`;
     let response = await this.urlLookup(advancedUrl, userPart, timeout);
     if (!response.buf && response.hasPolicy) {
-      return { keys: [], errs: [] }; // do not retry direct if advanced had a policy file
+      return [];
     }
     if (!response.buf) {
       response = await this.urlLookup(directUrl, userPart, timeout);
     }
     if (!response.buf) {
-      return { keys: [], errs: [] }; // do not retry direct if advanced had a policy file
+      return [];
+    }
+    if (typeof opgp !== 'undefined') {
+      return await KeyUtil.parseAndArmorKeys(response.buf);
     }
-    return await KeyUtil.readMany(response.buf);
+    // in pgp-block.html there is no openpgp loaded for performance, use background
+    const armored = await BrowserMsg.send.bg.await.pgpKeyBinaryToArmored({ binaryKeysData: response.buf });
+    return armored.keys;
   };
 
   public lookupEmail = async (email: string): Promise<PubkeysSearchResult> => {
-    const { keys, errs } = await this.rawLookupEmail(email);
-    if (errs.length) {
-      return { pubkeys: [] };
-    }
-    const pubkeys = keys.filter(key => key.emails.some(x => x.toLowerCase() === email.toLowerCase()));
-    if (!pubkeys.length) {
-      return { pubkeys: [] };
-    }
-    try {
-      return { pubkeys: pubkeys.map(pubkey => KeyUtil.armor(pubkey)) };
-    } catch (e) {
-      return { pubkeys: [] };
+    const all = await this.rawLookupEmail(email);
+    console.log(all);
+    const filtered = all.filter(key => key.emails.some(e => this.pubkeyUidFilter(e, email)));
+    return { pubkeys: filtered.map(pubkey => pubkey.armored) };
+  };
+
+  private pubkeyUidFilter = (uidEmail: string, expectedEmail: string) => {
+    if (!IS_MOCK_TEST_ENVIRONMENT) { // lgtm [js/trivial-conditional]
```

Comment:
I'd suggest to use normal email addresses in Wkd queries, but use this `IS_MOCK_TEST_ENVIRONMENT` flag in `rawLookupEmail` method instead to redirect queries to `localhost:8001` for certain emails (or for all emails except for the only one when we're testing the live Wkd?) -- line `const directHost = ...` .
Please also note this property `public port: number | undefined;`.
I created it earlier to hack into Wkd's behaviour from `test/source/tests/browser-unit-tests/unit-Wkd.js` tests. Now (if we change this line `const directHost = ...` accordingly) we'll be able to rely on `IS_MOCK_TEST_ENVIRONMENT` flag for these tests too and remove the `port` property.

Reply:
I'll take this suggestion but I'll do it a bit differently. Instead of `IS_MOCK_TEST_ENVIRONMENT` I'll use a property like `FORCE_WKD_HOST = ''` (prod) or `FORCE_WKD_HOST = 'http://localhost:8001'` (mock). It will be a bit more flexible, to be able to have one mock API per test (which I'd like to, eventually)